src/services/core/services-api/src/main/java/org/geoserver/geofence/services/RuleAdminService.java

```diff
@@ -33,6 +35,17 @@
 
     long insert(Rule rule, InsertPosition position);
 
+    /**
+     * @param rules to be inserted
+     * @return map of inserted rules along with their ids.
+     */
+    default Map<Long, Rule> insert(Set<Rule> rules) {
+        for (Rule rule: rules) {
+            insert(rule);
+        }
+        return Collections.emptyMap();
```

Comment:
instead of empty we could return an hashmap with real data. we have the id returned by the insertion method.